frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart

```diff
@@ -85,66 +93,121 @@ class DesktopGridTabBarBuilderImpl implements DatabaseTabBarItemBuilder {
 }
 
 class GridPage extends StatefulWidget {
-  final DatabaseController databaseController;
   const GridPage({
+    super.key,
     required this.view,
     required this.databaseController,
     this.onDeleted,
-    super.key,
+    this.initialRowId,
   });
 
   final ViewPB view;
+  final DatabaseController databaseController;
   final VoidCallback? onDeleted;
+  final String? initialRowId;
 
   @override
   State<GridPage> createState() => _GridPageState();
 }
 
 class _GridPageState extends State<GridPage> {
+  late final GridBloc _gridBloc;
+  bool _didOpenInitialRow = false;
+
   @override
   void initState() {
     super.initState();
+    _gridBloc = GridBloc(
+      view: widget.view,
+      databaseController: widget.databaseController,
+    )..add(const GridEvent.initial());
   }
 
   @override
   Widget build(BuildContext context) {
     return MultiBlocProvider(
       providers: [
-        BlocProvider<GridBloc>(
-          create: (context) => GridBloc(
-            view: widget.view,
-            databaseController: widget.databaseController,
-          )..add(const GridEvent.initial()),
+        BlocProvider<NotificationActionBloc>.value(
+          value: getIt<NotificationActionBloc>(),
         ),
+        BlocProvider<GridBloc>.value(value: _gridBloc),
```

Comment:
I prefer BlocProvider and use context.read because you won't forget to call close. If you want to keep this, this needs a close call in the dispose override

Reply:
Yeah, good point. Still a lot to do, and I need to do cleaning on all changes 😂👌